packages/lib/services/synchronizer/syncInfoUtils.ts

```diff
@@ -240,6 +240,26 @@ export class SyncInfo {
 		};
 	}
 
+	public filterSyncInfo() {
+		const filtered = JSON.parse(JSON.stringify(this));
+
+		// Filter content and checksum properties from master keys
+		if (filtered.masterKeys_) {
+			filtered.masterKeys_ = filtered.masterKeys_.map((mk: MasterKeyEntity) => {
+				delete mk.content;
+				delete mk.checksum;
+				return mk;
+			});
+		}
+
+		// Truncate the private key and public key
+		if (filtered.ppk_.value) {
+			filtered.ppk_.value.privateKey.ciphertext = `${filtered.ppk_.value.privateKey.ciphertext.substr(0, 20)}...${filtered.ppk_.value.privateKey.ciphertext.substr(-20)}`;
+			filtered.ppk_.value.publicKey = `${filtered.ppk_.value.publicKey.substr(0, 20)}...${filtered.ppk_.value.publicKey.substr(-20)}`;
```

Comment:
Not what was asked. The point of including the first 40 characters is to show the start of the public key data. With your change all we get is "-----BEGIN RSA PUBLI... RSA PUBLIC KEY-----" which is useless.

Reply:
sorry for the misunderstanding, have corrected it